src/NUnitEngine/nunit.engine.core/Internal/TestAssemblyResolver.cs

```diff
@@ -64,8 +77,74 @@ private Assembly OnResolving(AssemblyLoadContext context, AssemblyName name)
                 }
             }
 
+            foreach(string frameworkDirectory in AdditionalFrameworkDirectories)
+            {
+                var versionDir = FindBestVersionDir(frameworkDirectory, name.Version);
+                if (versionDir != null)
+                {
+                    string candidate = Path.Combine(frameworkDirectory, versionDir, name.Name + ".dll");
+                    if (File.Exists(candidate))
+                        return _loadContext.LoadFromAssemblyPath(candidate);
+                }
+            }
+
             return null;
         }
+
+        private static string GetDotNetInstallDirectory()
+        {
+            if (Path.DirectorySeparatorChar == '\\')
```

Comment:
Can we use `RuntimeInformation.IsOSPlatform(OSPlatform.Windows)` instead to check whether we are on windows?